runtime/src/fips.rs

```diff
@@ -37,55 +36,122 @@ impl FipsModule {
         }
         env.persistent_data.get_mut().zeroize();
     }
+}
 
-    /// Execute KAT for cryptographic algorithms implemented in H/W.
-    fn execute_kats(env: &mut Drivers) -> CaliptraResult<()> {
-        cprintln!("[kat] Executing SHA2-256 Engine KAT");
-        Sha256Kat::default().execute(&mut env.sha256)?;
-
-        cprintln!("[kat] Executing SHA2-384 Engine KAT");
-        Sha384Kat::default().execute(&mut env.sha384)?;
-
-        cprintln!("[kat] Executing SHA2-384 Accelerator KAT");
-        Sha384AccKat::default().execute(&mut env.sha384_acc)?;
+#[cfg(feature = "fips_self_test")]
+pub mod fips_self_test_cmd {
+    use super::*;
+    use crate::RtBootStatus::{RtFipSelfTestComplete, RtFipSelfTestStarted};
+    use caliptra_common::{verifier::FirmwareImageVerificationEnv, FMC_ORG, RUNTIME_ORG};
+    use caliptra_drivers::ResetReason;
+    use caliptra_image_verify::ImageVerifier;
+    use zerocopy::AsBytes;
+
+    // Helper function to create a mutable slice from a memory region
+    unsafe fn create_slice(org: u32, size: usize) -> &'static mut [u8] {
```

Comment:
As other parts of the firmware take  &'static references to values in .rodata (which is inside the returned slice), returning &mut here is unsound. Please return &'static [u8] instead and use core::slice::from_raw_parts(ptr, size) below.

Since FMC_ORG and RUNTIME_ORG are in ICCM; an NMI would be raised if anybody were to write to these slices :)

Reply:
Done. Yes these didn't need to be mutable.